src/Helpers/Retry.ts

```diff
@@ -0,0 +1,23 @@
+export type RetryCallbackProps = {
+  attempt: number;
+};
+
+export class Retry {
+  constructor(private maxRetries = 2, private retryDelay = 1000) {}
+
+  async fetch<T = unknown>(
+    callback: (data: RetryCallbackProps) => Promise<T> | T,
+  ) {
+    for (let attempt = 0; attempt < this.maxRetries; attempt++) {
+      try {
+        const data = callback({ attempt });
+        return data;
+      } catch {
+        console.log(`Retrying in ${this.retryDelay / 1000} seconds...`);
+        await new Promise((resolve) => setTimeout(resolve, this.retryDelay));
```

Comment:
If the rate limit of Github API is exceeded or the TOKEN is incorrect, the catch process will not start and will not be retried.
Because the callback does not throw an error.

Reply:
That's the magic :)
only callback will Throw, not the fetch.

PS: I forget to await response hehe